stabbity.lic

```diff
@@ -0,0 +1,253 @@
+        when 'is already quite dead'
+          pause 2
+        end
+        using_right_weapon?
+        if @thrown_weapon_npcs.include? @target
+          echo "*** Target #{@target} is in thrown_weapon_npcs ***"
+          kill_thrown
+          waitrt?
+          loot_mob unless @noloot
+          use_weapon 'preferred'
+          hide?
+        else
+          kill_stabbity
+          waitrt?
+          loot_mob unless @noloot
+          use_weapon 'preferred'
+          hide?
+        end
+      else
+        echo "*** No targets ***" if @debug
+        exit if single_mode?
+        pause 1
+      end
+    end
+  end
+
+  def kill_stabbity
+    while @npc_is_alive
+      fix_standing
+      case bput('backstab', 'You must be hidden to blindside', 
+                            'It would help if you were closer',
+                            'You\'ll need to stand up first',
+                            'flying out of reach',
+                            'to stop flying before',
+                            'Roundtime',
+                            /\[You\'re /,
+                            'There is nothing else')
+      when 'There is nothing else'
+        @npc_is_alive = false
+        return
+      when 'You must be hidden to blindside'
+        hide?
+        waitrt?
+      when 'It would help if you were closer'
+        fput('advance')
+        pause 3
+      when 'You\'ll need to stand up first'
+        fix_standing
+        hide?
+      when 'flying out of reach', 'to stop flying before'
+        echo "*** Switch to thrown -- add this MOB to your use_thrown_on list ***" if @debug
+        use_weapon 'thrown'
+        kill_thrown
+        return
+      when 'Roundtime', /\[You\'re /
+        return if npc_dead?
+        waitrt?
+        dodge_arena_trap if arena_mode?
+        hide?
+      end
+    end
+  end
+
+  def kill_thrown
+    action = @settings['stabbity']['thrown_action'].nil? ? 'throw' : @settings['stabbity']['thrown_action']
+    recover_command = @settings['stabbity']['thrown_invoke'] ? 'invoke' : "get #{@thrown_weapon}"
+    fix_standing
+    while @npc_is_alive
+      case bput(action, 'Roundtime', 'What are you trying', 'There is nothing else')
+      when 'What are you trying'
+        bput(recover_command, 'You pick up', 'suddenly leaps')
+        waitrt?
+      when 'Roundtime'
+        waitrt?
+        bput(recover_command, 'You pick up', 'suddenly leaps')
+        return if npc_dead?
+        waitrt?
+      when 'There is nothing else'
+        @npc_is_alive = false
+        return
+      end
+    end
+  end
+
+  def get_npcs
+    DRRoom.npcs - @ignore_npcs
+  end
+
+  def npc_dead?
+    if DRRoom.dead_npcs.include?(@target) || get_npcs.empty?
+      echo "*** Target #{@target} has died ***" if @debug
+      @npc_is_alive = false
+      return true
+    else
+      return false
+    end
+  end
+
+  def using_right_weapon?
```

Comment:
This function name should not end in a `?` if you aren't going to return a boolean and use its value

Reply:
Renamed to `switch_weapon` in next release.